### ISSUE TITLE
Incorrect traitlet keys passed via kwargs does not raise an error

### ISSUE THREAD

**Issue by watsonjj** [cta-observatory/ctapipe#699]
When configuring traitlets via constructor of the class (by passing the key and value via the kwargs), if an invalid traitlet is specified, no warning or error is raised. This has been identified as a problem in #697 and #661.

The problem lies in the traitlet package. When setting a traitlet via kwargs, this block of code is entered inside traitlets/traitlets.py:HasTraits

```python
    def __init__(self, *args, **kwargs):
        # Allow trait values to be set using keyword arguments.
        # We need to use setattr for this to trigger validation and
        # notifications.
        super_args = args
        super_kwargs = {}
        with self.hold_trait_notifications():
            for key, value in kwargs.items():
                if self.has_trait(key):
                    setattr(self, key, value)
                else:
                    # passthrough args that don't set traits to super
                    super_kwargs[key] = value
        try:
            super(HasTraits, self).__init__(*super_args, **super_kwargs)
        except TypeError as e:
            arg_s_list = [ repr(arg) for arg in super_args ]
            for k, v in super_kwargs.items():
                arg_s_list.append("%s=%r" % (k, v))
            arg_s = ', '.join(arg_s_list)
            warn(
                "Passing unrecoginized arguments to super({classname}).__init__({arg_s}).\n"
                "{error}\n"
                "This is deprecated in traitlets 4.2."
                "This error will be raised in a future release of traitlets."
                .format(
                    arg_s=arg_s, classname=self.__class__.__name__,
                    error=e,
                ),
                DeprecationWarning,
                stacklevel=2,
            )
```

If the traitlet does not exist then `except TypeError as e:` is entered, but the `warn` does not actually produce any output to the user (most likely a bug). It would seem that they intend to raise an error if the traitlet does not exist in the future, which would be ideal for us.

**Comment by MaxNoe** [comment]
Not a bug. DeprecationWarnings in python are silent by Design decision unless you ein `python -W`.

Is traitlets 4.2 out yet?

**Comment by watsonjj** [comment]
The traitlet version I am using is 4.3.2.

It would seem strange to have this warning passed silently. Is there anything we can do to make this warning appear even if the user is not using `python -W`?

**Comment by kosack** [comment]
As a hack, we could just add our own check in `Component.__init__()` that does the right thing, rather than waiting for traitlets to catch up.

**Comment by MaxNoe** [comment]
> It would seem strange to have this warning pass silently. I

This was a deliberate design decision by the python team. The reasoning was, that `DeprecationWarnings` are for developers and not for users. So ignoring them by default makes sense in that context.

https://docs.python.org/3.6/library/warnings.html#warning-categories

**Comment by watsonjj** [comment]
> This was a deliberate design decision by the python team. The reasoning was, that DeprecationWarnings are for developers and not for users. So ignoring them by default makes sense in that context.

Sorry, I meant that it seems a weird decision by the traitlets team to have the message `"Passing unrecoginized arguments to super({classname}).__init__({arg_s}).\n"` attached to the `DeprecationWarning` and therefore passes silently unless you are using `python -W`

**Comment by watsonjj** [comment]
Fixed by #701
